### PR TITLE
Convert dht doc to dns packet

### PR DESCRIPTION
**Pull request by diehuxx**
The last step in converting between did:dht documents and DNS packets. This includes:
1. Converting verification methods and verification relationships to and from DNS records and references thereof within the root record.
2. Converting services to and from DNS records and references references thereof within the root record
3. Converting alsoKnownAs to and from DNS records
4. Converting Controllers to and from DNS records

Possibly in follow up PR: More test coverage for `Document::to_packet` and `Document::from_packet` for unhappy paths.

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]
awesome work!

File: crates/dids/src/methods/dht/document_packet/mod.rs [hunk 1]
```diff
@@ -1,26 +1,439 @@
...
+            .find(|record| AlsoKnownAs::is_aka_record(record));
+        let also_known_as = match also_known_as {
+            None => None,
+            Some(aka) => Some(AlsoKnownAs::from_resource_record(aka)?),
+        };
+
+        // 6. Create document
+        Ok(Document {
+            id: did_uri,
+            context: None,
```
Comment (nitro-neal) [inline crates/dids/src/methods/dht/document_packet/mod.rs:310]: Is context: None on purpose?

File: crates/dids/src/methods/dht/document_packet/mod.rs [hunk 2]
```diff
@@ -1,26 +1,439 @@
...
+        VerificationMethod {
+            id: format!("{}#{}", did_uri, thumbprint),
+            r#type: "JsonWebKey".to_string(),
+            controller: did_uri.to_string(),
+            public_key_jwk: public_jwk.clone(),
+        }
+    }
+
+    #[test]
+    fn test_to_and_from_packet_kitchen_sink() {
```
Comment (nitro-neal) [inline crates/dids/src/methods/dht/document_packet/mod.rs:369]:
```suggestion
    fn test_to_and_from_packet_full_featured() {
```

**Review by nitro-neal — APPROVED** [review]
This is amazing, great work!

File: crates/dids/src/methods/dht/document_packet/mod.rs (hunk 1 above)
Comment (diehuxx) [reply crates/dids/src/methods/dht/document_packet/mod.rs:310]: Yep, it is. `did:dht` does not use `context`

 https://did-dht.com/#note-5
Screenshot 2024-06-03 at 12 36 47 PM=

**Pull request merged**